cstar/roms/runtime_settings.py

```diff
@@ -0,0 +1,806 @@
+
+    def __repr__(self) -> str:
+        """Return a full debug-style string representation of the settings.
+
+        Returns
+        -------
+        str
+            A detailed summary of the object, suitable for debugging.
+        """
+        attrs = {
+            "title": self.title,
+            "time_stepping": dict(self.time_stepping),
+            "bottom_drag": dict(self.bottom_drag),
+            "initial": dict(self.initial),
+            "forcing": [str(f) for f in self.forcing],
+            "output_root_name": self.output_root_name,
+            "grid": str(self.grid) if self.grid else None,
+            "climatology": str(self.climatology) if self.climatology else None,
+            "s_coord": dict(self.s_coord) if self.s_coord else None,
+            "rho0": self.rho0,
+            "lin_rho_eos": dict(self.lin_rho_eos) if self.lin_rho_eos else None,
+            "marbl_biogeochemistry": dict(self.marbl_biogeochemistry)
+            if self.marbl_biogeochemistry
+            else None,
+            "lateral_visc": self.lateral_visc,
+            "gamma2": self.gamma2,
+            "tracer_diff2": self.tracer_diff2
+            if self.tracer_diff2 is not None
+            else None,
+            "vertical_mixing": dict(self.vertical_mixing)
+            if self.vertical_mixing
+            else None,
+            "my_bak_mixing": dict(self.my_bak_mixing) if self.my_bak_mixing else None,
+            "sss_correction": self.sss_correction,
+            "sst_correction": self.sst_correction,
+            "ubind": self.ubind,
+            "v_sponge": self.v_sponge,
+        }
+
+        inner = ", ".join(f"{k}={repr(v)}" for k, v in attrs.items() if v is not None)
+        return f"{self.__class__.__name__}({inner})"
+
+    def to_file(self, filepath: PathLike[str]) -> None:
+        """Write the current settings to a ROMS-compatible `.in` file.
+
+        Parameters
+        ----------
+        filepath : str or Path
+            Path where the output file will be written.
+        """
+
+        filepath = Path(filepath)
+
+        output_order = [
+            "title",
+            "time_stepping",
+            "bottom_drag",
+            "initial",
+            "forcing",
+            "output_root_name",
+            "s_coord",
+            "grid",
+            "marbl_biogeochemistry",
+            "lateral_visc",
+            "rho0",
+            "lin_rho_eos",
+            "gamma2",
+            "tracer_diff2",
+            "vertical_mixing",
+            "my_bak_mixing",
+            "sss_correction",
+            "sst_correction",
+            "ubind",
+            "v_sponge",
+            "climatology",
+        ]
+
+        with filepath.open("w") as f:
+            for field in output_order:
+                if getattr(self, field) is None:
+                    continue
+                fieldname = self.model_fields[field].alias or field
+                f.write(self.model_dump(by_alias=True)[fieldname])
```

Comment:
This method should be implemented as a custom serializer to enable using `pydantic` to do the heavy lifting. 

If you need a separate view of the object for this file, create another class to provide exactly that view. That would also avoid the hardcoded list of attribute names.